lib/client.js

```diff
@@ -53,8 +54,12 @@ function Client(config = {}) {
   if (config.version) {
     this.version = config.version;
   }
-
-  this.connectionSettings = cloneDeep(config.connection || {});
+  // if the type of connection settings is a function, pass it to the next layer
```

Comment:
Why is this commented out?

Reply:
conniectionProvider takes the responsibility to manage the connection settings.

https://github.com/tgriesser/knex/pull/3403/files#diff-cbc1ae6887774cf34c8d4bc73dece12fR16

Reply:
yeah, but why
```
  // if (typeof config.connection === 'function') {
  //   this.connectionSettings = config.connection;
  // } else {
  //   this.connectionSettings = cloneDeep(config.connection || {});
  // }
```
is committed to the repo?

Reply:
I've removed the comments